sound/soc/sof/core.c

```diff
@@ -405,6 +405,9 @@ static int sof_probe_continue(struct snd_sof_dev *sdev)
 	if (plat_data->sof_probe_complete)
 		plat_data->sof_probe_complete(sdev->dev);
 
+	/* initialize default D0 sub-state */
+	sdev->d0_substate = SOF_DSP_D0I0;
+
```

Comment:
I would do the initialization earlier, since the sof_probe_complete is where you typically enable pm_runtime